### ISSUE TITLE
OAuthApplicationDatabase update_application infinite recursion

### ISSUE THREAD

**Issue by forslund** [OpenVoiceOS/selene_api#60]
```
>>> db.update_application(**current)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  File "/home/ake/.venvs/ovos/lib/python3.11/site-packages/ovos_backend_client/database.py", line 425, in update_application
    self.update_application(oauth_service,
  [Previous line repeated 996 more times]
RecursionError: maximum recursion depth exceeded
```

The [update_application()](https://github.com/OpenVoiceOS/ovos-backend-client/blob/dev/ovos_backend_client/database.py#L421) method calls itself causing above exception.

What is the intention of this method? should it be forwarded to the standard update method or should it do additional checks?

**Comment by JarbasAl** [comment]
i think that was meant to just call add_application internally? def not intended to be like that

**Comment by forslund** [comment]
Resolved in #62 , closing
